pkg/server/server.go

```diff
@@ -376,30 +429,42 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 	cycleDetectionCheckResolver.SetDelegate(localChecker)
 	localChecker.SetDelegate(cycleDetectionCheckResolver)
 
-	if s.dispatchThrottlingCheckResolverEnabled {
+	if s.checkDispatchThrottlingEnabled {
+		s.logger.Info("Enabling Check dispatch throttling",
+			zap.Duration("Frequency", s.checkDispatchThrottlingFrequency),
+			zap.Uint32("DefaultThreshold", s.checkDispatchThrottlingDefaultThreshold),
+			zap.Uint32("MaxThreshold", s.checkDispatchThrottlingMaxThreshold),
+		)
+
 		dispatchThrottlingConfig := graph.DispatchThrottlingCheckResolverConfig{
-			Frequency:        s.dispatchThrottlingCheckResolverFrequency,
-			DefaultThreshold: s.dispatchThrottlingDefaultThreshold,
-			MaxThreshold:     s.dispatchThrottlingMaxThreshold,
+			DefaultThreshold: s.checkDispatchThrottlingDefaultThreshold,
+			MaxThreshold:     s.checkDispatchThrottlingMaxThreshold,
 		}
 
-		if s.dispatchThrottlingMaxThreshold != 0 && s.dispatchThrottlingDefaultThreshold > s.dispatchThrottlingMaxThreshold {
+		if s.checkDispatchThrottlingMaxThreshold != 0 && s.checkDispatchThrottlingDefaultThreshold > s.checkDispatchThrottlingMaxThreshold {
 			return nil, fmt.Errorf("default dispatch throttling threshold must be equal or smaller than max dispatch threshold")
 		}
 
-		s.logger.Info("Enabling dispatch throttling",
-			zap.Duration("Frequency", s.dispatchThrottlingCheckResolverFrequency),
-			zap.Uint32("DefaultThreshold", s.dispatchThrottlingDefaultThreshold),
-			zap.Uint32("MaxThreshold", s.dispatchThrottlingMaxThreshold),
+		dispatchThrottlingCheckResolver := graph.NewDispatchThrottlingCheckResolver(
+			graph.WithDispatchThrottlingCheckResolverConfig(dispatchThrottlingConfig),
+			graph.WithThrottler(throttler.NewConstantRateThrottler(s.checkDispatchThrottlingFrequency, "check_dispatch_throttle")),
 		)
-
-		dispatchThrottlingCheckResolver := graph.NewDispatchThrottlingCheckResolver(dispatchThrottlingConfig)
 		dispatchThrottlingCheckResolver.SetDelegate(localChecker)
 		s.dispatchThrottlingCheckResolver = dispatchThrottlingCheckResolver
 
 		cycleDetectionCheckResolver.SetDelegate(dispatchThrottlingCheckResolver)
 	}
 
+	if s.listObjectsDispatchThrottlingEnabled {
+		s.logger.Info("Enabling ListObjects dispatch throttling",
+			zap.Duration("Frequency", s.listObjectsDispatchThrottlingFrequency),
+			zap.Uint32("DefaultThreshold", s.listObjectsDispatchDefaultThreshold),
+			zap.Uint32("MaxThreshold", s.listObjectsDispatchThrottlingMaxThreshold),
+		)
+
```

Comment:
You're missing the same validation checks that we have above for the Check dispatch throttling. Namely, if `listObjectsDispatchThrottling.threshold` is greater than `listObjectsDispatchThrottling.maxThreshold`, then it should return an error.

Please also add a test for this in `server_test.go`. The server constructor `NewServerWithOpts` should return an error for this scenario.